// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
syntax = "proto2";

package arolla.serialization_codecs;

import "arolla/serialization_base/base.proto";

option cc_enable_arenas = true;

message DictV1Proto {
  extend arolla.serialization_base.ValueProto {
    optional DictV1Proto extension = 326033766;
  }

  // Represents KeyToRowDictQType.
  //
  // ValueProto.input_value_indices[0]
  //   -- key qtype.
  message KeyToRowDictQTypeProto {}

  // Represents DictQType.
  //
  // ValueProto.input_value_indices[0]
  //   -- key qtype.
  // ValueProto.input_value_indices[1]
  //   -- value qtype.
  message DictQTypeProto {}

  oneof value {
    KeyToRowDictQTypeProto key_to_row_dict_qtype = 101;
    DictQTypeProto dict_qtype = 102;
  }
}
